instrumentation/dalli/lib/opentelemetry/instrumentation/dalli/utils.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    # Utility functions
+    module Utils
+      extend self
+
+      STRING_PLACEHOLDER = ''.encode(::Encoding::UTF_8).freeze
+      CMD_MAX_LEN = 500
+
+      OPNAME_MAPPING = {
+        'get'           => 'get',
+        'cas'           => 'get',
+        'set'           => 'set',
+        'add'           => 'add',
+        'replace'       => 'replace',
+        'delete'        => 'delete',
+        'incr'          => 'incr',
+        'decr'          => 'decr',
+        'flush'         => 'flush',
+        'write_noop'    => 'noop',
+        'version'       => 'version',
+        'send_multiget' => 'getkq',
+        'append'        => 'append',
+        'prepend'       => 'prepend',
+        'stats'         => 'stat',
+        'reset_stats'   => 'stat',
+        'multi_set'     => 'setq',
+        'multi_add'     => 'addq',
+        'multi_replace' => 'replaceq',
+        'multi_delete'  => 'deleteq',
+        'touch'         => 'touch',
+        # 'sasl_authentication' => 'auth_negotiation',
+        # 'sasl_authentication' => 'auth_request',
```

Comment:
There are two underlying write requests for the `sasl_authentication` so I'm not sure if it should map to one or the other here.

Reply:
maybe we just add a heuristic in the `opname` method to make this choice, and return like `'auth'` from opname mapping? my thought is it's better to be accurate than necessarily have the tightest code
```
if (temp_opname = OPNAME_MAPPING[lookup_name] || operation.to_s) && temp_opname == 'auth'
  // additional heuristic to check if its negotiation or request
else
  return temp_opname
end
```

Reply:
I'm kinda OK with handling this case later. As it is, the default in `opname` will catch this and turn it into `"sasl_authentication"` in both cases.

Reply:
Yeah trying to dig into the `sasl_authentication` method here https://github.com/petergoldstein/dalli/blob/master/lib/dalli/server.rb#L701-L730 and I'm not sure how I could extract more meaningful info out of it right now.  This puts us in a better place to iterate on it going forward when/if needed.

Reply:
works for me 👍